vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -323,21 +323,16 @@ def get_moe_configs(E: int, N: int,
     return None
 
 
-def get_default_config(
-    M: int,
-    E: int,
-    N: int,
-    K: int,
-    topk: int,
-    dtype: Optional[str],
-) -> Dict[str, int]:
+def get_default_config(M: int, E: int, N: int, K: int, topk: int,
+                       dtype: Optional[str],
+                       is_marlin: bool) -> Dict[str, int]:
     config = {
         'BLOCK_SIZE_M': 64,
         'BLOCK_SIZE_N': 64,
         'BLOCK_SIZE_K': 32,
         'GROUP_SIZE_M': 8
     }
-    if M <= E:
+    if M <= E or (is_marlin and M <= 32):
```

Comment:
Is `M <= 32` because we dequant fully outside of that case?

Reply:
@ElizaWszola

Reply:
It's a heuristic aimed at performance: marlin kernels are faster with this blocking for `M <= 32`